centreon/src/Core/ResourceAccess/Application/UseCase/DeleteRule/DeleteRule.php

```diff
@@ -0,0 +1,148 @@
+    {
+        /**
+         * Check if current user is authorized to perform the action.
+         * Only users linked to AUTHORIZED_ACL_GROUPS acl_group and having access in Read/Write rights on the page
+         * are authorized to add a Resource Access Rule.
+         */
+        if (! $this->isAuthorized()) {
+            $this->error(
+                "User doesn't have sufficient rights to delete a resource access rule",
+                [
+                    'user_id' => $this->user->getId(),
+                ]
+            );
+            $presenter->setResponseStatus(
+                new ForbiddenResponse(RuleException::notAllowed()->getMessage())
+            );
+
+            return;
+        }
+
+        try {
+            $this->info('Starting resource access rule deletion', ['rule_id' => $ruleId]);
+            if (! $this->readRepository->exists($ruleId)) {
+                $this->error('Resource access rule not found', ['rule_id' => $ruleId]);
+                $presenter->setResponseStatus(new NotFoundResponse('Resource access rule'));
+
+                return;
+            }
+
+            $this->deleteRule($ruleId);
+            $presenter->setResponseStatus(new NoContentResponse());
+        } catch (\Throwable $exception) {
+            $presenter->setResponseStatus(new ErrorResponse(RuleException::errorWhileDeleting($exception)));
+            $this->error($exception->getMessage(), ['trace' => $exception->getTraceAsString()]);
+        }
+    }
+
+    /**
+     * @return bool
+     */
+    private function isAuthorized(): bool
+    {
+        $userAccessGroupNames = array_map(
+            static fn (AccessGroup $accessGroup): string => $accessGroup->getName(),
+            $this->accessGroupRepository->findByContact($this->user)
+        );
+
+        /**
+         * User must be
+         *     - An admin (belongs to the centreon_admin_acl ACL group)
+         *     - authorized to reach the Resource Access Management page.
+         */
+        return ! (empty(array_intersect($userAccessGroupNames, self::AUTHORIZED_ACL_GROUPS)))
+            && $this->user->hasTopologyRole(Contact::ROLE_ADMINISTRATION_ACL_RESOURCE_ACCESS_MANAGEMENT_RW);
+    }
+
+    private function deleteRule(int $ruleId): void
+    {
+        $this->debug('Starting transaction');
+        $this->dataStorageEngine->startTransaction();
+
+        try {
+            /**
+             * Here are the deletions (on cascade or not) that will occur on rule deletion
+             *     - Contact relations (ON DELETE CASCADE)
+             *     - Contact Group relations (ON DELETE CASCADE)
+             *     - Datasets relations + datasets (NEED MANUAL DELETION)
+             *     - DatasetFilters (ON DELETE CASCADE).
+             */
+            $datasetIds = $this->readRepository->findDatasetIdsByRuleId($ruleId);
+            $this->writeRepository->delete($ruleId);
+            $this->writeRepository->deleteDatasets($datasetIds);
```

Comment:
Can datasets exist without a rule?
If not, you can delete datasets directly in the delete method.

Reply:
Yes they can exist without a rule that is why I've decided to delete it aside

Reply:
Seen together. Changes were made.